README.md

```diff
@@ -1,14 +1,64 @@
 Foodcoops.net deployment demo
 =============================
 
-You need to create a certificate for your setup before you can start. For testing purpose only you can add it via `COPY` in `haproxy/Dockerfile` or you mount it via an volume in the `docker-compose.yml`. Check out the comments the files.
+You need to create a certificate for your setup before you can start. For testing purpose only you can add it via `COPY` in `haproxy/Dockerfile` or you mount it via an volume in the `docker-compose.yml`. Check out the comments the files (also see [section below](#Generating_test_certificates)).
 
 To get it running you need to provide the private information via environment variables to `docker-compose`. Here is an example to build and start the project:
 
 ```shell
 export FOODSOFT_DB_PASSWORD=foodsoft
-export FOODSOFT_SECRET_KEY_BASE=secret
+export FOODSOFT_SECRET_KEY_BASE=secret_fs
+export SHAREDLISTS_DB_PASSWORD=sharedlists
+export SHAREDLISTS_SECRET_KEY_BASE=secret_sl
 export MYSQL_ROOT_PASSWORD=mysql
+
 docker-compose build --pull
 docker-compose up -d
 ```
+
+## Initial database setup
+
+On first time run, you'll need to setup the database. Start and connect to it as root:
+
+```shell
+docker-compose up -d mariadb redis
+docker inspect foodcoopsnet_mariadb_1 | grep '"IPAddress"'
+# "IPAddress": "172.20.0.2",
+mysql -h 172.20.0.2 -u root -p
+```
+
+Then run the following SQL commands:
+
+```sql
+CREATE DATABASE foodsoft CHARACTER SET utf8 COLLATE utf8_bin;
+GRANT ALL ON foodsoft.* TO foodsoft@'%' IDENTIFIED BY 'secret_fs';
+
+CREATE DATABASE sharedlists CHARACTER SET utf8 COLLATE utf8_bin;
+GRANT ALL ON sharedlists.* TO sharedlists@'%' IDENTIFIED BY 'secret_sl';
+GRANT SELECT ON sharedlists.* TO foodsoft@'%';
+```
+
+Finally you need to populate the databases:
+
+```shell
+docker-compose run --rm foodsoft_web bundle exec rake db:setup
```

Comment:
Here I got stuck because of the existing (empty) database:

> Mysql2::Error: Can't create database 'foodsoft'; **database exists**: [...]

However, this might again be my fault because I am a newbie: another mysterious (related?) point is that the target IP reads 172.19.0.4:

> root@foodcoopsnet:~/foodcoops.net# docker-compose run --rm foodsoft_web bundle exec rake db:setup
> `/nonexistent` is not a directory.
> Bundler will use `/tmp/bundler/home/unknown' as your home directory temporarily.
> Access denied for user 'foodsoft'@'**172.19.0.4**' (using password: YES)Please provide the root password for your MySQL installation

However, I think the correct IP should be 172.19.0.3 in my case:
> root@foodcoopsnet:\~/foodcoops.net# docker inspect foodcoopsnet_mariadb_1 | grep '"IPAddress"'
>            "IPAddress": "",
>                    "IPAddress": "**172.19.0.3**",
> root@foodcoopsnet:~/foodcoops.net# docker-compose run --rm foodsoft_web ping mariadb
> PING **mariadb (172.19.0.3)**: 56 data bytes
> 64 bytes from 172.19.0.3: icmp_seq=0 ttl=64 time=0.089 ms

The .4 seems to belong to the localhost where `rake db:setup` is running:
> root@foodcoopsnet:~/foodcoops.net# docker-compose run --rm foodsoft_web **ip addr**
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
>    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
>    inet 127.0.0.1/8 scope host lo
>       valid_lft forever preferred_lft forever
> 35: eth0@if36: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
>    link/ether 02:42:ac:13:00:04 brd ff:ff:ff:ff:ff:ff
>    inet **172.19.0.4**/16 scope global eth0
>       valid_lft forever preferred_lft forever

Is the `DATABASE_URL` correctly passed into the container?

(Sorry - only questions and no help from my side (yet)).

Reply:
Hmmm, what if you do `rake db:schema:load db:seed` instead? That avoids creating the db.
Can you access it manually using `mysql` with the same database url settings?

But indeed, if the IP is off, that's another matter. Strange, can't see why yet.

(And no problem - I think it's important to have these things sorted out, or else no-one else will be able to join easily.)

Reply:
Finally, it worked for me. `db:setup` was not the problem - it just notices and prints `foodsoft already exists` before creating and seeding the tables. The problem in my case was about the passwords. I will comment on that line.

Moreover, I got some insight to the IP issue. I think the target IP (for the mariadb) container is correctly resolved, but `foodsoft@172.19.0.4` (.4 being the foodsoft_web IP) means "connecting **from** foodsoft_web".

I was able to login now. I like the setup.